### PR TITLE
Initial Vapor setup

### PR DESCRIPTION
**Pull request by Kamelon7**

### CODE REVIEW

File: .github/workflows/hub-deploy-sandbox.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Require Vapor CLI
+        run: composer global require laravel/vapor-cli
+
+      - name: Install Project Dependencies
+        run: composer install --no-interaction --prefer-dist --no-dev --optimize-autoloader
+
+      - name: Run PHP artisan
+        run: php artisan event:cache
+
+      - name: Install NPM
```
Comment (emmachughes) [inline .github/workflows/hub-deploy-sandbox.yaml:33]: This step is [redundant](https://docs.npmjs.com/cli/v9/commands/npm-ci) to `npm ci` below.

File: sourcecode/hub/production.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+FROM laravelphp/vapor:php82
+
+COPY . /var/task
```
Comment (emmachughes) [inline sourcecode/hub/production.Dockerfile:3]: Each line is supposed to be terminated with a line feed, according to the rules in `.editorconfig`. This doesn't make much difference for YAML files, but it matters for e.g. shell scripts, so might want to ensure your editor abides by `.editorconfig`.

File: .github/workflows/hub-deploy-sandbox.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          php-version: 8.2
+          tools: composer:v2
+          coverage: none
+      
+      - name: Require Vapor CLI
+        run: composer global require laravel/vapor-cli
+
+      - name: Install Project Dependencies
+        run: composer install --no-interaction --prefer-dist --no-dev --optimize-autoloader
```
Comment (emmachughes) [inline .github/workflows/hub-deploy-sandbox.yaml:28]: Will these steps have any effect? This just installs dependencies and stuff in the CI environment. Doesn't `vapor deploy` ensure the same steps in `vapor.yml` are run in the image that ends up being deployed?

**Review by emmachughes — CHANGES_REQUESTED** [review]
Some workflow changes are needed.

File: .github/workflows/hub-deploy-sandbox.yaml (hunk 1 above)
Comment (Kamelon7) [reply .github/workflows/hub-deploy-sandbox.yaml:30]: Removed

File: sourcecode/hub/production.Dockerfile (hunk 2 above)
Comment (Kamelon7) [reply sourcecode/hub/production.Dockerfile]: These files were added by Vapor during initialization, which is why they didn't abide by `.editorconfig`.
I've added line feed terminations to both this file and `staging.Dockerfile`.

File: .github/workflows/hub-deploy-sandbox.yaml (hunk 3 above)
Comment (Kamelon7) [reply .github/workflows/hub-deploy-sandbox.yaml:28]: That sounds like a valid assumption, but the [Vapor documentation](https://docs.vapor.build/1.0/projects/deployments.html#example-with-github-actions) keeps these in their GH example, and in the other CI examples.

Comment (emmachughes) [reply .github/workflows/hub-deploy-sandbox.yaml:28]: I think vapor-cli interacts with Laravel in some ways, so this is probably why it's done that way. The event caching and npm build can probably be removed from the workflow, then.

Comment (Kamelon7) [reply .github/workflows/hub-deploy-sandbox.yaml:28]: As in, remove the whole line `npm ci && npm run build && rm -rf node_modules`, or just the `npm build` part?

Comment (emmachughes) [reply .github/workflows/hub-deploy-sandbox.yaml:28]: The whole line

**Review by emmachughes — APPROVED** [review]

**Pull request merged**